src/Bicep.Core/FileSystem/FileResolver.cs

```diff
@@ -236,5 +241,35 @@ private static void RequireFileUri(Uri uri)
                 throw new ArgumentException($"Non-file URI is not supported by this file resolver.");
             }
         }
+
+        private static void ApplyWindowsConFileWorkaround(string localPath)
+        {
+            /*
+             * Win10 (and possibly older versions) will block without returning when
+             * reading a file whose name is CON which breaks the language server
+             * 
+             * as a workaround, we will simulate Win11+ behavior that throws a
+             * FileNotFoundException
+             * 
+             * https://github.com/Azure/bicep/issues/6224
+             */
+
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                // not on windows - no need for the workaround
+                return;
+            }
+
+            string fileName = Path.GetFileName(localPath);
+            if(!string.Equals(fileName, "CON", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
@majastrz - this is just one of the cases that won't work - there's a bunch of reserved filenames https://stackoverflow.com/questions/32561117/get-reserved-file-names-in-windows

Reply:
The never returning behavior is specific to `CON` only. The others don't behave that way, so we're fine. However, I was not aware that you can't also have a file extension after `con`, so `con.txt` will cause the language server to freeze.